app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/ws/WsModels.scala

```diff
@@ -0,0 +1,75 @@
+package org.bitcoins.commons.jsonmodels.ws
+
+import org.bitcoins.commons.jsonmodels.bitcoind.GetBlockHeaderResult
+import org.bitcoins.core.api.wallet.db.SpendingInfoDb
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.crypto.StringFactory
+
+sealed trait WsType
+
+object WsType extends StringFactory[WsType] {
+
+  override def fromString(string: String): WsType = {
+    WalletWsType.fromString(string)
+  }
+}
+
+sealed trait WalletWsType extends WsType
+
+object WalletWsType extends StringFactory[WalletWsType] {
+  case object TxProcessed extends WalletWsType
+  case object TxBroadcast extends WalletWsType
+  case object ReservedUtxos extends WalletWsType
+  case object NewAddress extends WalletWsType
+  case object BlockProcessed extends WalletWsType
+
+  private val all =
+    Vector(TxProcessed, TxBroadcast, ReservedUtxos, NewAddress, BlockProcessed)
+
+  override def fromStringOpt(string: String): Option[WalletWsType] = {
+    all.find(_.toString.toLowerCase() == string.toLowerCase)
+  }
+
+  override def fromString(string: String): WalletWsType = {
+    fromStringOpt(string)
+      .getOrElse(sys.error(s"Cannot find wallet ws type for string=$string"))
+  }
+}
+
+sealed trait WsPushNotification[T] {
+  def `type`: WalletWsType
+  def payload: T
+}
+
+sealed trait WalletNotification[T] extends WsPushNotification[T] {
```

Comment:
seems kind of strange to have notification extend push notification, are there any non-push notifications? I'm not sure I know what the intended difference is between these two types is

Reply:
I renamed `WsPushNotification` -> `WsNotification`. 

I have `WalletNotification` because the event originates from the wallet project. We could add `ChainNotification` and `NodeNotification` etc in the future for those respective projects.